src/jorphan/org/apache/jorphan/math/HistogramStatCalculatorLong.java

```diff
@@ -0,0 +1,194 @@
+            sum += histoCalc.sum;
+            bytes += histoCalc.bytes;
+            sentBytes += histoCalc.sentBytes;
+            histogram.add(histoCalc.histogram);
+            max = Math.max(histoCalc.max, max);
+            min = Math.min(histoCalc.min, min);
+            valuesMap = updateAllValueCount(valuesMap, histoCalc.valuesMap);
+        } else {
+            throw new IllegalArgumentException("Only instances of HistogramStatCalculator allowed.");
+        }
+    }
+
+    @Override
+    public Long getMedian() {
+        return histogram.getValueAtPercentile(50)/1000000;
+    }
+
+    @Override
+    public long getTotalBytes() {
+        return bytes;
+    }
+
+    @Override
+    public long getTotalSentBytes() {
+        return sentBytes;
+    }
+
+    @Override
+    public Long getPercentPoint(float percent) {
+        return getPercentPoint((double) percent);
+    }
+
+    @Override
+    public Long getPercentPoint(double percent) {
+        return histogram.getValueAtPercentile(100.0 * percent)/1000000;
+    }
+
+    @Override
+    public Map<Number, Number[]> getDistribution() {
+        Map<Number, Number[]> result = new HashMap<>();
+        for (Map.Entry<Long,Long> entry : valuesMap.entrySet()) {
+            Number[] dis = new Number[2];
+            dis[0] = entry.getKey();
+            dis[1] = entry.getValue();
+            result.put(entry.getKey(), dis);   
+        }
+        return result;
+    }
+
+    @Override
+    public double getMean() {
+        return histogram.getMean()/1000000;
+    }
+
+    @Override
+    public double getStandardDeviation() {
+        return histogram.getStdDeviation()/1000000;
+    }
+
+    @Override
+    public Long getMin() {
+        return min;
+    }
+
+    @Override
+    public Long getMax() {
+        return max;
+    }
+
+    @Override
+    public long getCount() {
+        return histogram.getTotalCount();
+    }
+
+    @Override
+    public double getSum() {
+        return sum;
+    }
+    /**
+     * @return the valuesMap
+     */
+    public Map<Long, Long> getValuesMap() {
+        return valuesMap;
+    }
+
+
+    @Override
+    public void addValue(Long val, long sampleCount) {
+        sum += val * sampleCount;
+        for (int i = 0; i < sampleCount; i++) {
+            latencyStats.recordLatency(val*1000000);
+            latencyStats.getIntervalHistogramInto(intervalHistogram);
+            histogram.add(intervalHistogram);
+        }
+        max = Math.max(val, max);
+        min = Math.min(val, min);
+        updateValueCount(val,sampleCount);
+    }
+
+    @Override
+    public void addValue(Long val) {
+        sum += val;
+        latencyStats.recordLatency(val*1000000);
+        latencyStats.getIntervalHistogramInto(intervalHistogram);
+        histogram.add(intervalHistogram);
+        max = Math.max(val, max);
+        min = Math.min(val, min);
+        updateValueCount(val,1);
+    }
+    private void updateValueCount(Long actualValue, long sampleCount) {
```

Comment:
This is sad. Is it really required?

Reply:
It is a method for calculating distribution, so I try to record the value when it is added . Can you give another idea? Thanks

Reply:
What's the purpose of `valuesMap`?
Histogram / LatencyStats should track the distribution.

Reply:
In fact in the old code, the  distribution value comes from this method : 
```
public Map<Number, Number[]> getDistribution() {
        Map<Number, Number[]> items = new HashMap<>();

        for (Entry<T, MutableLong> entry : valuesMap.entrySet()) {
            Number[] dis = new Number[2];
            dis[0] = entry.getKey();
            dis[1] = entry.getValue();
            items.put(entry.getKey(), dis);
        }
        return items;
    }
```
I don't think the LatencyStats can track this distribution